src/array.rs

```diff
@@ -984,12 +1000,15 @@ impl<D: Dimension> PyArray<PyObject, D> {
     ///         }).unwrap(),
     ///     ];
     ///
-    ///     let pyarray = PyArray::from_owned_object_array(py, array);
+    ///     let pyarray = PyArray::from_owned_object_array_bound(py, array);
     ///
     ///     assert!(pyarray.readonly().as_array().get(0).unwrap().as_ref(py).is_instance_of::<CustomElement>());
     /// });
     /// ```
-    pub fn from_owned_object_array<'py, T>(py: Python<'py>, mut arr: Array<Py<T>, D>) -> &'py Self {
+    pub fn from_owned_object_array_bound<T>(
+        py: Python<'_>,
+        mut arr: Array<Py<T>, D>,
```

Comment:
I do wonder whether there is something to be discussed whether this can and/or should take `Array<Bound<'_, T>, D>`? It should be possible but I also do not really see a benefit from doing so?

Reply:
Hmm, interesting point. I also don't really see a clear benefit of doing so either.

Reply:
Let's walk the straight and narrow here and limit ourselves to the necessary API change without making it more complicated than it has to be.